comp/core/flare/flare.go

```diff
@@ -73,11 +79,9 @@ func newFlare(deps dependencies) (provides, rcclienttypes.TaskListenerProvider)
 		diagnoseDeps: diagnoseDeps,
 	}
 
-	endpoint := EndpointProvider{flareComp: f}
-
 	p := provides{
 		Comp:     f,
-		Endpoint: api.NewAgentEndpointProvider(endpoint, "/flare", "POST"),
+		Endpoint: api.NewAgentEndpointProvider(f.createAndReturnFlarePath, "/flare", "POST"),
```

Comment:
just a note for multiple endpoints since that was the original pain here, the value groups don't expand an array of providers I've found so you will still need to list them all in the provides struct

```
provides{
    Comp: f,
    EndpointOne: ...,
    EndpointTwo: ...,
}
```

Reply:
We might be able to circumvent that using [value group flattening ](https://pkg.go.dev/go.uber.org/fx#hdr-Value_group_flattening)

Reply:
Maybe I'm misunderstanding but it seems it's possible to flatten array of value group (https://github.com/uber-go/dig/pull/264)

```golang
// Value groups can be used to provide multiple values for a group from a
// dig.Out using slices, however considering groups are retrieved by requesting
// a slice this implies that the values must be retrieved using a slice of
// slices. As of dig v1.9.0, if you want to provide individual elements to the
// group instead of the slice itself, you can add the `flatten` modifier to the
// group from a dig.Out.
//
//   type IntResult struct {
//     dig.Out
//
//     Handler []int `group:"server"`         // [][]int from dig.In
//     Handler []int `group:"server,flatten"` // []int from dig.In
//   }
//
```
from https://github.com/uber-go/dig/pull/264/files#diff-a20b1b3b4b2bca5bef2b853a6e3f19def513381f9c7ee68d6979f0435c885dcf

Did you tried this ?

Reply:
Maybe I'm misunderstanding but it seems it's possible to flatten slice of value group providers (https://github.com/uber-go/dig/pull/264).

An extract of the [update documentation file](https://github.com/uber-go/dig/pull/264/files#diff-a20b1b3b4b2bca5bef2b853a6e3f19def513381f9c7ee68d6979f0435c885dcf):
```golang
//
// Value groups can be used to provide multiple values for a group from a
// dig.Out using slices, however considering groups are retrieved by requesting
// a slice this implies that the values must be retrieved using a slice of
// slices. As of dig v1.9.0, if you want to provide individual elements to the
// group instead of the slice itself, you can add the `flatten` modifier to the
// group from a dig.Out.
//
//   type IntResult struct {
//     dig.Out
//
//     Handler []int `group:"server"`         // [][]int from dig.In
//     Handler []int `group:"server,flatten"` // []int from dig.In
//   }
//
```
I haven't tried it, so I don't know what it's worth.